app/generate.js

```diff
@@ -97,8 +107,24 @@ function generatePages(renderedComponents) {
       page.conclusionHtml = '';
     }
 
-    page.header = header;
-    page.footer = footer;
+    if (!page.header) {
+      page.header = defaultHeader;
+    } else {
+      if (!page.header.logo) {
```

Comment:
This `if` is supposed to move into the `generateHeader` function. The general idea is to avoid duplicating logic as much as possible (this greatly helps you to avoid bugs).